src/main/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtil.java

```diff
@@ -0,0 +1,559 @@
+                    String exceptionMessage = (String) mapBasedInstallKernel.get("action.error.message");
+                    debug(exceptionMessage);
+                }
+            }
+        } catch (PrivilegedActionException e) {
+            throw new PluginExecutionException("Could not load the jar " + getMapBasedInstallKernelJar(installDirectory).getAbsolutePath(), e);
+        }
+    }
+    
+    private Map<String, Object> createMapBasedInstallKernelInstance(File installDirectory) throws PrivilegedActionException {
+        final File installJarFile = getMapBasedInstallKernelJar(installDirectory);
+        String installJarFileSubpath = installJarFile.getParentFile().getName() + File.separator + installJarFile.getName();
+        Map<String, Object> mapBasedInstallKernel = AccessController.doPrivileged(new PrivilegedExceptionAction<Map<String, Object>>() {
+            @SuppressWarnings({ "unchecked", "resource" })
+            @Override
+            public Map<String, Object> run() throws Exception {
+                ClassLoader loader = new URLClassLoader(new URL[] { installJarFile.toURI().toURL() }, null);
+                Class<Map<String, Object>> clazz;
+                clazz = (Class<Map<String, Object>>) loader.loadClass("com.ibm.ws.install.map.InstallMap");
+                return clazz.newInstance();
+            }
+        });
+        if (mapBasedInstallKernel == null){
+            debug("mbik is null");
+        }
+
+        // Init
+        mapBasedInstallKernel.put("runtime.install.dir", installDirectory);
+        mapBasedInstallKernel.put("install.map.jar", installJarFileSubpath);
+        debug("install.map.jar: " + installJarFileSubpath);
+        debug("install.kernel.init.code: " + mapBasedInstallKernel.get("install.kernel.init.code"));
+        debug("install.kernel.init.error.message: " + mapBasedInstallKernel.get("install.kernel.init.error.message"));
+        File usrDir = new File(installDirectory, "usr/tmp");
+        mapBasedInstallKernel.put("target.user.directory", usrDir);
+        return mapBasedInstallKernel;
+    }
+    
+    /**
+     * Find latest install map jar from lib directory
+     * 
+     * @return the install map jar file
+     */
+    private File getMapBasedInstallKernelJar(File installDirectory) {
+        final String installMapPrefix = "com.ibm.ws.install.map";
+        final String installMapSuffix = ".jar";
+
+        File dir = new File(installDirectory, "lib");
+
+        File[] installMapJars = dir.listFiles(new FilenameFilter() {
+            @Override
+            public boolean accept(File dir, String name) {
+                return name.startsWith(installMapPrefix) && name.endsWith(installMapSuffix);
+            }
+        });
+
+        File latestJar = null;
+        if (installMapJars != null && installMapJars.length > 0) {
+            for (File jar : installMapJars) {
```

Comment:
The logic in this loop is very confusing. There are a bunch of solutions online on how to do this. See [here](https://stackoverflow.com/questions/198431/how-do-you-compare-two-version-strings-in-java) for an example.